### PR TITLE
refactor: CC => FC

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | refactor: CC => FC |
| 🇨🇳 Chinese | refactor: CC => FC |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by li-jia-nan** [comment]
@zombieJ @MadCcc @afc163 这个 CI 挂的有点奇怪，外面覆盖率不够，但是点进去里面又是正常的

**Comment by zheeeng** [comment]
image

神奇

File: components/spin/index.tsx [hunk 1]
```diff
@@ -75,79 +70,38 @@ function shouldDelay(spinning?: boolean, delay?: number): boolean {
...
+    className,
+    size = 'default',
+    tip,
+    wrapperClassName,
+    style,
+    children,
+    ...restProps
+  } = props;
+
+  const [spinning, setSpinning] = React.useState<boolean>(
```
Comment (zombieJ) [inline components/spin/index.tsx:87]: 这边可以直接用 useMergedState 了

Comment (li-jia-nan) [reply components/spin/index.tsx:87]: useMergedState 不支持防抖，这里先用 useState

**Comment by afc163** [comment]
https://app.codecov.io/gh/ant-design/ant-design/blob/master/components/affix/index.tsx

图片

这个覆盖率要在 master 先补上。

**Comment by li-jia-nan** [comment]
> https://app.codecov.io/gh/ant-design/ant-design/blob/master/components/affix/index.tsx
> 
> 图片
> 
> 这个覆盖率要在 master 先补上。

看了下这一块代码，这个不应该没测到，上面有几个case都测到了，我再研究一下



> https://app.codecov.io/gh/ant-design/ant-design/blob/master/components/affix/index.tsx
> 
> 图片
> 
> 这个覆盖率要在 master 先补上。

- #37993

File: components/spin/index.tsx [hunk 2]
```diff
@@ -75,79 +70,38 @@ function shouldDelay(spinning?: boolean, delay?: number): boolean {
...
+    () => customSpinning && !shouldDelay(customSpinning, delay),
+  );
 
-  constructor(props: SpinClassProps) {
-    super(props);
+  const updateSpinning = debounce<() => void>(() => {
```
Comment (zombieJ) [inline components/spin/index.tsx:91]: 这里只有 effect 里用，感觉是可以直接放 effect 里的？否则每次 render 都创建一个 debounce wrapper function

Comment (li-jia-nan) [reply components/spin/index.tsx]: done

**Review by zombieJ — APPROVED** [review]

**Pull request merged**